### ISSUE TITLE
Using headerBackButtonDisplayMode: "minimal" causes back button to jump across screen

### ISSUE THREAD

**Issue by eeston** [expo/expo#33812]
### Minimal reproducible example

https://github.com/eeston/expo-router-back-button-issue

### Which package manager are you using? (Yarn is recommended)

npm

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

None

### Summary

When using `headerBackButtonDisplayMode: "minimal"` along with `headerLargeTitle: true`, the back button jumps across the screen.

Video here: https://github.com/eeston/expo-router-back-button-issue/blob/main/recording.mp4

It is more obvious when you run on a device as the back button jumps as soon as you press the side of the screen (as if you were to start to swipe back).

### Environment

```text
expo-env-info 1.2.1 environment info:
    System:
      OS: macOS 15.2
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 22.10.0 - ~/.nvm/versions/node/v22.10.0/bin/node
      Yarn: 1.22.22 - ~/.nvm/versions/node/v22.10.0/bin/yarn
      npm: 10.9.0 - ~/.nvm/versions/node/v22.10.0/bin/npm
      Watchman: 2024.10.28.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.2, iOS 18.2, macOS 15.2, tvOS 18.2, visionOS 2.2, watchOS 11.2
    IDEs:
      Android Studio: 2024.1 AI-241.18034.62.2412.12266719
      Xcode: 16.2/16C5032a - /usr/bin/xcodebuild
    npmPackages:
      expo: ~52.0.20 => 52.0.21
      expo-router: ~4.0.14 => 4.0.14
      react: 18.3.1 => 18.3.1
      react-dom: 18.3.1 => 18.3.1
      react-native: 0.76.5 => 0.76.5
      react-native-web: ~0.19.13 => 0.19.13
    npmGlobalPackages:
      eas-cli: 12.6.2
    Expo Workflow: managed
```

**Comment by eeston** [comment]
https://github.com/user-attachments/assets/f5ee4862-0874-4b54-bd5c-21ae1c1e1c8e

**Comment by marklawlor** [comment]
Hey! Thanks for opening the issue. Seems that this issue is related to `react-navigation` library which is a dependency of Expo Router. Can you also post your issue in [this repo](https://github.com/react-navigation/react-navigation) so that it's notified to the maintainers of that library? This will help us fix the issue faster since it's up to the maintainers of that library to investigate it.

**Comment by jerone** [comment]
Already reported: https://github.com/react-navigation/react-navigation/issues/11946

**Comment by iliapnmrv** [comment]
Hi! react-native-screens fixed this issue in the latest update 
```json
"@react-navigation/native": "7.1.4",
"@react-navigation/native-stack": "7.3.8",
"react-native-screens": "4.11.0-beta.0",
```
these versions work fine for me

**Comment by eeston** [comment]
Thanks for the heads up @iliapnmrv. Will check it out 🙌

**Comment by jerone** [comment]
This requires those packages to be updated in Expo, right?

**Comment by iliapnmrv** [comment]
@jerone Hi! I'm not sure how expo packages work exactly. But as I understand, when you install expo packages, you also install `react-navigation` and `react-native-screens` packages. And I guess only updating these packages to latest versions would be enough

**Comment by jerone** [comment]
> [@jerone](https://github.com/jerone) Hi! I'm not sure how expo packages work exactly. But as I understand, when you install expo packages, you also install `react-navigation` and `react-native-screens` packages. And I guess only updating these packages to latest versions would be enough

Yeah, but Expo Doctor seems to not like that. Those packages need to be updated in https://github.com/expo/expo/blob/sdk-53/packages/expo/bundledNativeModules.json

**Comment by krystofwoldrich** [comment]
The `react-native-screens` package is up to date (latest 4.11.1) in https://github.com/expo/expo/blob/b1ae25877ad792fba26442f7cce0585e52ca39a2/packages/expo/bundledNativeModules.json#L103

What `expo` and `expo-router` versions are you using?

**Comment by jerone** [comment]
Confirmed fixed for me.

**Comment by eeston** [comment]
Closing this as it has seems to be working for myself and others now.
Thanks all.

**Comment by gorkabalm** [comment]
It's happening again with iOS 26 :(
